Formula/berkeley-db.rb

```diff
@@ -9,7 +9,7 @@ class BerkeleyDb < Formula
 
   livecheck do
     url "https://www.oracle.com/database/technologies/related/berkeleydb-downloads.html"
-    regex(%r{href=.*?/berkeley-db/db[._-]v?(\d+(?:\.\d+)+)\.t}i)
+    regex(/h3>Berkeley\s*DB.*\(\s*(\d+(?:\.\d+)+)\s*\)/i)
```

Comment:
```suggestion
    regex(/Berkeley\s*DB[^(]*?\(\s*v?(\d+(?:\.\d+)+)\s*\)/i)
```

A couple small tweaks here:

* I'm leaning towards simply omitting the `h3` tag from the regex altogether, as matching currently works fine without it (i.e., it matches what we want without matching anything we don't want). My concern is that your regex above would break if it changes to something other than an `h3` element in the future. My thinking is that the looseness is potentially beneficial in this particular context and we can always revisit the regex to make it more explicit if it matches more than it should in the future. [As an aside, if we did want to include the `h3` element in the regex, we would want to use `<h3[^>]*?>`, which would continue to work if attributes are added to the element in the future.]
* We should use the contextually-appropriate `[^(]*?` instead of the loose and greedy `.*`, which could match more than we expect if the page format unexpectedly changes

Reply:
Both sound great, on it!

Curious: how does `livecheck` regex work if it finds multiple matches within the HTML?

Reply:
> Curious: how does `livecheck` regex work if it finds multiple matches within the HTML?

The specifics can sometimes depend on the strategy but in most cases: the capture group is extracted from each match, converted to a [`Version`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/version.rb) object, and then they're compared against one another to find the greatest value. For example, if we had 1.2.2, 1.2.3, and 1.2.4 matches on the page, 1.2.4 would be treated as the newest. The order of the matches on the page doesn't matter and duplicate matches also don't make a difference.

`Version` comparison typically works fine but it can be a problem when string tokens are compared. For example, if we have `1.2.3-c05955ce` and `1.2.3-f5bb9bad` versions, the suffix would be compared using string comparison and this doesn't always yield the correct newest version. It may be that the `c05955ce` version is actually newer, so it would be a problem for `f5bb9bad` to be treated as newest from the perspective of `Version` comparison. We're usually able to address these types of situations in contextually-appropriate ways but it's just something to remember about how `Version` comparison works.

There's a little more to this process in certain contexts but that's the important bits. `Version` sorting is something that livecheck does internally but if we need more control over how we're matching versions (before livecheck works with them), we can use a `strategy` block. You'll see these in some formulae that need a more advanced approach (i.e. something we can't do with a regex) and they vary from simple (e.g., manipulating version strings) to complex. There are various common patterns in `strategy` blocks and they tend to be a modified version of what the strategy is already doing internally (i.e., we only stray from default behavior as needed).

Hopefully this answers your question but let me know if any of it is unclear.